glean-core/src/metrics/counter.rs

```diff
@@ -24,4 +25,19 @@ impl CounterMetric {
                 _ => Metric::Counter(amount),
             })
     }
+
+    /// **Test-only API (exported for FFI purposes).**
+    ///
+    /// Get the currently stored value as an integer.
+    ///
+    /// This doesn't clear the stored value.
+    pub fn test_get_value(&self, glean: &Glean, storage_name: &str) -> Option<u64> {
+        let snapshot = StorageManager.snapshot_as_json(glean.storage(), storage_name, false);
+        snapshot
+            .as_object()
+            .and_then(|o| o.get("counter"))
+            .and_then(|o| o.as_object())
+            .and_then(|o| o.get(&self.meta.identifier()))
+            .and_then(|o| o.as_i64().map(|i| i as u64))
```

Comment:
In the future we support deserialization in the `Metrics` enum so we could do this generically?  (Not to hold up this PR)

Reply:
We might, I need to find a way to make that nice.